cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -199,6 +202,39 @@ func generateInitramfsMounts() (err error) {
 	return nil
 }
 
+func canInstallAndRunAtOnce(mst *initramfsMountsState) (bool, error) {
+	currentSeed, err := mst.LoadSeed(mst.recoverySystem)
+	if err != nil {
+		return false, err
+	}
+	preseedSeed, ok := currentSeed.(seed.PreseedCapable)
+	if !ok {
+		return false, nil
+	}
+
+	// TODO: relax this condition when "install and run" well tested
+	if !preseedSeed.HasArtifact("preseed.tgz") {
+		return false, nil
+	}
+
+	// If kernel has fde-setup hook, then we should also have fde-setup in initramfs
+	kernelPath := filepath.Join(boot.InitramfsRunMntDir, "kernel")
+	kernelHasFdeSetup := osutil.FileExists(filepath.Join(kernelPath, "meta", "hooks", "fde-setup"))
+	_, fdeSetupErr := exec.LookPath("fde-setup")
+	if kernelHasFdeSetup && fdeSetupErr != nil {
+		return false, nil
+	}
+
+	gadgetPath := filepath.Join(boot.InitramfsRunMntDir, "gadget")
+	if osutil.FileExists(filepath.Join(gadgetPath, "meta", "hooks", "install-device")) {
+		return false, nil
+	}
+
+	// TODO: turn it into
```

Comment:
Could you please clarify this TODO? Maybe specifying the conditions under which we can install and run the system immediately as part of the function description would help to understand this.

Reply:
Fixed as part of 5f3453adf2e88c4e9b9917f2bb9b054f67b67e80